### PR TITLE
EVG-6917 display tasks shouldn't have dependencies

### PR DESCRIPTION
**Pull request by ablack12**
Spent a really long time looking for this thing that John had actually linked in the ticket.

### CODE REVIEW

File: model/lifecycle.go [hunk 1]
```diff
@@ -852,9 +852,8 @@ func createTasksForBuild(project *Project, buildVariant *BuildVariant, b *build.
 			}
 		}
 
-		// Display tasks depend on all exec task dependencies
+		// Local pointer to display task
 		if displayTask, ok := displayTasks[newTask.DisplayName]; ok {
-			displayTask.DependsOn = append(displayTask.DependsOn, newTask.DependsOn...)
```
Comment (john-m-liu) [inline model/lifecycle.go]: @ybrill was this something needed for EVG-5960?

Comment (ybrill) [reply model/lifecycle.go]: I was recently wondering the same thing 😄 
I remember it fixed something, but I don't remember what. Display tasks don't have dependencies and other tasks can't depend on them, so it seems strange to give them dependencies.

One thing that comes to mind is a display task is shown as blocked in the UI when any of its exec tasks are blocked. By adding dependencies to the display task we don't need to check the blocked status of each of the exec tasks on page load.

Comment (ablack12) [reply model/lifecycle.go]: Maybe to minimize risk I'll just change this to not have duplicates rather than removing dependencies entirely.

**Comment by ablack12** [comment]
This commit is a little messy but there's potentially a reason we started giving display tasks dependencies (potentially to show as blocked, as Jonathan mentioned). If we don't think time is a concern, rather than creating displayTaskInfo to cache dependency information early on, we could create a new function when handling display tasks around 864 that's just displayTask.HasDependency(taskId) or something.

File: model/lifecycle.go [hunk 2]
```diff
@@ -768,10 +772,14 @@ func createTasksForBuild(project *Project, buildVariant *BuildVariant, b *build.
 		if err != nil {
 			return tasks, errors.Wrapf(err, "Failed to create display task %s", id)
 		}
+		dependencyIds := map[string]bool{}
```
Comment (john-m-liu) [inline model/lifecycle.go]: I feel like I'm missing something, but isn't t set by the call to createDisplayTask above, which doesn't set the dependencies? So I don't think this block does anything

Comment (ablack12) [reply model/lifecycle.go]: okay whoa true

**Pull request merged**